src/functions.php

```diff
@@ -45,11 +45,17 @@ function nullify(mixed $value, ?string $cast = null): mixed
 function extract_last_event_id(ServerRequestInterface $request): ?string
 {
     $qs = query_string($request->getUri(), new FlatQueryParser());
+    $lastEventId = nullify($request->getHeaderLine('Last-Event-ID')) ?? $qs->getParam('lastEventID');
+    if (null === $lastEventId) {
+        $lastEventId = $qs->getParam('Last-Event-ID') ??
```

Comment:
Not directly related to this patch but a new helper method allowing to retrieve the query parameter in a case insensitive way would be more reliable.

Reply:
yes, i was wondering whether we should support case-insensitive keys in mercure? because currently lasteventid=foo wouldn't be taken into account.

Reply:
IMHO the current implementation is enough. Query parameters aren't case insensitive.